### PR TITLE
Fixing rank 1 outputs for WordPieceTokenizer

### LINKED ISSUE #49: Fix rank 1 outputs for WordPieceTokenizer
Right now, a scalar string inputs to the tokenizer will turn into rank 2 outputs, which we do not want. Scalar inputs should result in rank 1 outputs.

See this colab for a reproduction of the issue:
https://colab.research.google.com/gist/mattdangerw/8e1fdbebadf0ff6ff2dc065b2e00db65/rank1wordpiece.ipynb

### PR DESCRIPTION
**Pull request by aflah02**
Fixes #49 
I've added a unit test as well based on the recommendations to test with batching tensors with `tf.data.experimental.dense_to_ragged_batch`
Sorry for the late PR I got winded up in some exams

### CODE REVIEW

File: keras_nlp/tokenizers/word_piece_tokenizer.py [hunk 1]
```diff
@@ -256,6 +256,15 @@ def get_config(self) -> Dict[str, Any]:
         return config
 
     def tokenize(self, inputs):
+        # Check if Input is Scalar or Not
+        if not isinstance(inputs, tf.Tensor) or not isinstance(
```
Comment (mattdangerw) [inline keras_nlp/tokenizers/word_piece_tokenizer.py:260]: I think we could just convert to tensor generally here, rather then converting only to check the rank. Add something like this first...

```
        if not isinstance(inputs, (tf.Tensor, tf.RaggedTensor)):
            inputs = tf.convert_to_tensor(inputs)
```

Then you can just run `scalar_input = tf.convert_to_tensor(inputs).shape.rank == 0` assuming tensor.

File: keras_nlp/tokenizers/word_piece_tokenizer.py [hunk 2]
```diff
@@ -282,6 +291,11 @@ def tokenize(self, inputs):
             output_shape = tokens.shape.as_list()
             output_shape[-1] = self._sequence_length
             tokens = tokens.to_tensor(shape=output_shape)
+        # Convert to a dense output if input in scalar
+        if scalar_input:
+            tokens = tf.squeeze(tokens, 0)
+            tf.ensure_shape(tokens, shape=[None])
```
Comment (mattdangerw) [inline keras_nlp/tokenizers/word_piece_tokenizer.py:297]: Does the unit test below break without this? If it doesn't we might be able to remove.

If we do need it, we should probably update this set the shape to `[self._sequence_length]`. If sequence length is set, we do want to make sure the static shape is correct set to `[self._sequence_length]`.

**Review by mattdangerw — CHANGES_REQUESTED** [review]
Looks good! Thank you!

Just a few comments, then good to go I think!

File: keras_nlp/tokenizers/word_piece_tokenizer.py (hunk 1 above)
Comment (aflah02) [reply keras_nlp/tokenizers/word_piece_tokenizer.py:260]: Got it this is indeed a better way, thanks!

File: keras_nlp/tokenizers/word_piece_tokenizer.py (hunk 2 above)
Comment (aflah02) [reply keras_nlp/tokenizers/word_piece_tokenizer.py:297]: Yeah, it does fail without this check, I've made the changes to set shape appropriately now

**Comment by aflah02** [comment]
@mattdangerw Thanks for the review, I've made the changes as advised

**Review by mattdangerw — APPROVED** [review]
Thank you! Looks great.

**Pull request merged**